File: common/src/main/java/dev/obscuria/fragmentum/server/FragmentumServerRegistry.java
package dev.obscuria.fragmentum.server;

import com.mojang.brigadier.CommandDispatcher;
import dev.obscuria.fragmentum.Fragmentum;
import net.minecraft.commands.CommandBuildContext;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.commands.Commands;

@SuppressWarnings("unused")
public interface FragmentumServerRegistry
{
    static void registerCommand(CommandRegistrar registrar)
    {
        Fragmentum.SERVICES.server().registerCommand(registrar);
    }

    @FunctionalInterface
    interface CommandRegistrar
    {
        void register(
                CommandDispatcher<CommandSourceStack> dispatcher,
                CommandBuildContext registryAccess,
                Commands.CommandSelection environment);
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/server/FragmentumServerRegistry.kt
@file:Suppress("unused", "RedundantNullableReturnType")

package dev.obscuria.fragmentum.server

import com.mojang.brigadier.CommandDispatcher
import dev.obscuria.fragmentum.Fragmentum
import net.minecraft.commands.CommandBuildContext
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands

object FragmentumServerRegistry {

    fun registerCommand(registrar: CommandRegistrar) {
        Fragmentum.SERVICES.server().registerCommand(registrar)
    }

    fun interface CommandRegistrar {

        fun register(
            dispatcher: CommandDispatcher<CommandSourceStack>,
            registryAccess: CommandBuildContext,
            environment: Commands.CommandSelection
        )
    }
}